cedar-policy/src/api/err.rs

```diff
@@ -731,6 +731,30 @@ impl From<ast::UnexpectedSlotError> for PolicySetError {
     }
 }
 
+/// Represents one or more [`ParseError`]s encountered when parsing a policy or
+/// expression.
+/// By default, the `Diagnostic` and `Error` implementations will only print the
+/// first error. If you want to see all errors, use `.iter()` or `.into_iter()`.
+#[derive(Debug, Diagnostic, Error)]
+#[error(transparent)]
+#[diagnostic(transparent)]
+pub struct ParseErrors(#[from] cedar_policy_core::parser::err::ParseErrors);
+
+impl ParseErrors {
+    /// Get every [`ParseError`] associated with this [`ParseErrors`] object.
+    /// The returned iterator is guaranteed to be nonempty.
+    pub fn iter(&self) -> impl Iterator<Item = &ParseError> {
+        self.0.iter().map(ParseError::ref_cast)
+    }
+}
+
+/// Errors that can occur when parsing policies or expressions.
+#[derive(Debug, Diagnostic, Error, RefCast)]
+#[repr(transparent)]
+#[error(transparent)]
+#[diagnostic(transparent)]
+pub struct ParseError(#[from] cedar_policy_core::parser::err::ParseError);
```

Comment:
Regarding exposing `ParseError` as an enum in the future, we should fine based on https://predr.ag/blog/turning-rust-struct-to-enum-is-not-always-breaking/ (by the author of cargo-semver-checks, so I assume they know what they're talking about)